### PR TITLE
feat: submodule implementation

### PR DESCRIPTION
**Pull request by rubenVB01**
## Description

Implementation of the aws-kms submodule of the encryption-at-rest module.

### CODE REVIEW

File: modules/aws-kms/variables.tf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  nullable    = true
+}
+
+variable "aws_kms_key_arn" {
+  description = "The ARN of the AWS KMS key (e.g., arn:aws:kms:us-east-1:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab)."
+  type        = string
+}
+
+variable "kms_key_region" {
+  description = "The AWS region in which the AWS customer master key exists: CA_CENTRAL_1, US_EAST_1, US_EAST_2, US_WEST_1, US_WEST_2, SA_EAST_1."
```
Comment (EspenAlbert) [inline modules/aws-kms/variables.tf:26]: maybe we can add a validation:
  validation {
    condition = upper(var.kms_key_region) == var.kms_key_region
    error_message = "The value of kms_key_region must be uppercase."
  }

**Review by EspenAlbert — APPROVED** [review]
LGTM.
One thought I had.
If the user specifies this multiple times, then maybe:
`mongodbatlas_cloud_provider_access_authorization`
`mongodbatlas_cloud_provider_access_setup`
Needs to be conditional? I think `mongodbatlas_cloud_provider_access_setup` can only be done once per project.

However, if it is specified multiple times, maybe it is easier for the user to use the resources directly 🤔

**Comment by rubenVB01** [comment]
> LGTM. One thought I had. If the user specifies this multiple times, then maybe: `mongodbatlas_cloud_provider_access_authorization` `mongodbatlas_cloud_provider_access_setup` Needs to be conditional? I think `mongodbatlas_cloud_provider_access_setup` can only be done once per project.
> 
> However, if it is specified multiple times, maybe it is easier for the user to use the resources directly 🤔

If the user defines several times the module, wouldn't they share the same tfstate? If so, I guess the setup resource would not be re-created because it uses the same project id.

**Comment by rubenVB01** [comment]
> > LGTM. One thought I had. If the user specifies this multiple times, then maybe: `mongodbatlas_cloud_provider_access_authorization` `mongodbatlas_cloud_provider_access_setup` Needs to be conditional? I think `mongodbatlas_cloud_provider_access_setup` can only be done once per project.
> > However, if it is specified multiple times, maybe it is easier for the user to use the resources directly 🤔
> 
> If the user defines several times the module, wouldn't they share the same tfstate? If so, I guess the setup resource would not be re-created because it uses the same project id.

Screenshot 2024-08-14 at 13 37 12


Screenshot 2024-08-14 at 13 38 20

**Comment by EspenAlbert** [comment]
@rubenVB01 nice! Thank you for testing this.

**Pull request merged**